src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -26,6 +26,7 @@ internal abstract partial class MemberSemanticModel : CSharpSemanticModel
         // The bound nodes associated with a syntax node, from highest in the tree to lowest.
         private readonly Dictionary<SyntaxNode, ImmutableArray<BoundNode>> _guardedNodeMap = new Dictionary<SyntaxNode, ImmutableArray<BoundNode>>();
         private Dictionary<SyntaxNode, BoundStatement> _lazyGuardedSynthesizedStatementsMap;
+        private NullableWalker.SnapshotManager _lazyGuardedSnapshotManager;
```

Comment:
>_lazyGuardedSnapshotManager [](start = 47, length = 27)

Why do we need to guard access to the ```SnapshotManager```? isn't it immutable?

Reply:
It's immutable, but the field itself is not. Because it's tied to the nodes in the map and should be set at the same time from the same pass, we lock access to it to prevent races.

---
In reply to: [295519202](https://github.com/dotnet/roslyn/pull/36563#discussion_r295519202) [](ancestors = 295519202)